File: src/libs/ThirdParty.VBSuppliers/Properties/YouTravel/IYouTravelSettings.vb
Imports ThirdParty.Abstractions

Public Interface IYouTravelSettings
    ReadOnly Property AllowCancellations(tpAttributeSearch As IThirdPartyAttributeSearch, IsMandatory As Boolean) As Boolean
    ReadOnly Property Username(tpAttributeSearch As IThirdPartyAttributeSearch) As String
    ReadOnly Property Password(tpAttributeSearch As IThirdPartyAttributeSearch) As String
    ReadOnly Property SearchURL(tpAttributeSearch As IThirdPartyAttributeSearch) As String
    ReadOnly Property BookingURL(tpAttributeSearch As IThirdPartyAttributeSearch) As String
    ReadOnly Property CancellationFeeURL(tpAttributeSearch As IThirdPartyAttributeSearch) As String
    ReadOnly Property CancellationURL(tpAttributeSearch As IThirdPartyAttributeSearch) As String
    ReadOnly Property LangID(tpAttributeSearch As IThirdPartyAttributeSearch) As String
    ReadOnly Property OffsetCancellationDays(tpAttributeSearch As IThirdPartyAttributeSearch, IsMandatory As Boolean) As Integer
    ReadOnly Property UseGZip(tpAttributeSearch As IThirdPartyAttributeSearch) As Boolean
    ReadOnly Property PrebookURL(tpAttributeSearch As IThirdPartyAttributeSearch) As String
    ReadOnly Property CancellationPolicyURL(tpAttributeSearch As IThirdPartyAttributeSearch) As String
End Interface

File: src/libs/iVectorOne.Suppliers/YouTravel/IYouTravelSettings.cs
namespace iVectorOne.CSSuppliers.YouTravel
{
    public interface IYouTravelSettings
    {
        bool AllowCancellations(IThirdPartyAttributeSearch tpAttributeSearch, bool IsMandatory);
        string User(IThirdPartyAttributeSearch tpAttributeSearch);
        string Password(IThirdPartyAttributeSearch tpAttributeSearch);
        string SearchURL(IThirdPartyAttributeSearch tpAttributeSearch);
        string BookingURL(IThirdPartyAttributeSearch tpAttributeSearch);
        string CancellationFeeURL(IThirdPartyAttributeSearch tpAttributeSearch);
        string CancellationURL(IThirdPartyAttributeSearch tpAttributeSearch);
        string LanguageCode(IThirdPartyAttributeSearch tpAttributeSearch);
        int OffsetCancellationDays(IThirdPartyAttributeSearch tpAttributeSearch, bool IsMandatory);
        bool UseGZip(IThirdPartyAttributeSearch tpAttributeSearch);
        string PrebookURL(IThirdPartyAttributeSearch tpAttributeSearch);
        string CancellationPolicyURL(IThirdPartyAttributeSearch tpAttributeSearch);
    }
}